tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitProcessHandler.cs

```diff
@@ -106,43 +110,52 @@ public virtual CommandResult Run(string arguments, string workingDirectory)
                 Arguments = arguments
             };
 
-            try
+            if (!AssetTasks.TryGetValue(workingDirectory, out var value))
             {
-                DebugLogger.LogInformation($"git {arguments}");
-                var process = Process.Start(processStartInfo);
-                string stdOut = process.StandardOutput.ReadToEnd();
-                string stdErr = process.StandardError.ReadToEnd();
-                process.WaitForExit();
+                value = new TaskQueue();
+                // only add a new task queue if it doesn't exist
+                // if it does exist, just use the old one so that pending tasks will still be able to complete
+                AssetTasks.AddOrUpdate(workingDirectory, value, (key, oldValue) => oldValue);
+            }
```

Comment:
There's still a race condition here.  If AssetTasks.TryGetValue returns false, but AddOrUpdate hits the Update func, you'll keep the old value in the dictionary but continue to use the new value created on line 115 for the enqueue statement on  line 121

Reply:
You could rewrite using GetOrAdd like:

```cs
            // only add a new task queue if it doesn't exist
            var value = AssetTasks.GetOrAdd(workingDirectory, () => new TaskQueue());
```

Reply:
Thanks for the feedback. Definitely added.